### ISSUE TITLE
Variable Node in filename not resolved

### ISSUE THREAD

**Issue by dennis-zyska** [neilmunday/slurm-mail#61]
Versions

OS version: Centos 7
Slurm version: 22.05.02
Slurm Mail version: 4.1

Describe the bug

The variable Node in the log filename is not resolved, so that the log is not displayed correctly in the e-mail.

```
The last 25 lines of <path>_Training_.31224.%N.log are shown below:

slurm-mail: file<path>__Training.31224.%N.log does not exist
```

**Comment by neilmunday** [comment]
Nice spot.

This is a tricky one. Slurm-Mail is using the `scontrol` output to find the job output and error files. E.g.

```bash
scontrol show job 2 | grep -e StdErr -e StdOut
   StdErr=/root/foo.%N.out
   StdOut=/root/foo.%N.out
```

As you can see, `scontrol` does not expand the `%N` unfortunately.

If you override the names of your nodes then you can't rely on the allocated node name to work out the value of `%N`. For example, in my docker test set-up for Slurm, I override the name of the nodes from the docker hostname:

```bash
# scontrol show node
NodeName=node01 Arch=x86_64 CoresPerSocket=1
   CPUAlloc=0 CPUEfctv=1 CPUTot=1 CPULoad=0.43
   AvailableFeatures=(null)
   ActiveFeatures=(null)
   Gres=(null)
   NodeAddr=127.0.0.1 NodeHostName=localhost Version=22.05.5
   ...
```

When using `%N` in this set-up, this resolves to the hostname of the node that ran the job which happens to be `0044cd2c8b9e` rather than `node01`.

Additionally, there are other place holders than can be used in job names. See [sbatch man page](https://slurm.schedmd.com/sbatch.html#lbAH)

I think it might be best if Slurm-Mail checks for any place holders in the output and error file names and if present doesn't attempt to tail the file and gives the user a warning instead.

Thoughts?

**Comment by neilmunday** [comment]
Just to add, using `%j` is ok.

**Comment by neilmunday** [comment]
Testing of the allowed filename patterns, results as follows:

- [ ] %A
- [ ] %a
- [ ] %J
- [ ] %j
- [ ] %N
- [ ] %n
- [ ] %s
- [ ] %t
- [ ] %u
- [ ] %x

**Comment by neilmunday** [comment]
Note: see issue #67 for another bug relating to spaces in the name of the output file.

**Comment by neilmunday** [comment]
Fixed in 4.1 branch.
